ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,182 @@
+            LOGGER.error(
+                "Exception converting query response to JSON: "
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        """get the prometheus query result
+
+        Args:
+            query(str): promthetheus query string
+
+        Returns:
+            dict: query result
+        """
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_alert(self, alert_name):
+        """
+        Get alert by alert anem if it's an active alert
+
+        Args:
+             alert(str): alert name
+
+        Examples:
+             result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             dict: dictiionary containing alert name as key and alert data as value
+        """
+        response = self.alerts
+        matched_alerts = {}
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert["labels"]["alertname"] == alert_name:
+                alert_list.append(
+                    {
+                        "labels": alert["labels"],
+                        "annotations": alert["annotations"],
+                        "state": alert["state"],
+                        "activeAt": alert["activeAt"],
+                        "value": alert["value"],
+                    }
+                )
+        matched_alerts[alert_name] = alert_list
+        return matched_alerts
+
+    def alert_sampler(self, alert, timeout):
+        """
+        Sample output for an alert and raises timeout exception
+
+        Args:
+             alert(str): alert name
+             timeout(int): wait time, default is 10 mins
+
+        Raise:
+             TimeoutExpiredError: if alert is not fired before wait_timeout
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout or TIMEOUT_10MIN,
+            sleep=1,
+            func=self.get_alert,
+            alert=alert,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                for item in sample[alert]:
+                    if item["state"] == "firing":
+                        return
+        except TimeoutExpiredError:
+            LOGGER.error(
+                f"Failed to get successful alert {alert}. Current data: {sample[alert]} "
+            )
+            raise
+
+    def query_sampler(self, query, timeout):
+        """
+        Sample output for query function
+
+        Args:
+             query(str): prometheus query string
+             wait_timeout(int): default is 10 mins
+
+        Return:
+             list: return the query result
+
+        Raise:
+             TimeoutExpiredError: if query response doesn't return success
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout or TIMEOUT_10MIN,
+            sleep=1,
```

Comment:
For query, we should sleep based on prometheus scrapeInterval. Typically it is 30 seconds, but this is configurable in a cluster. There is no point making queries every 1/5 seconds, if the update is not supposed to happen for scrapeInterval seconds.

oc get prometheus -n openshift-monitoring -o json | jq .items[0].spec.scrapeInterval

Reply:
added function to get scrape interval